lib/test/api_test.rb

```diff
@@ -0,0 +1,259 @@
+require 'veritrans'
+require 'minitest/autorun'
+
+class TestVeritrans < Minitest::Test
+
+  def setup
+    @mt_test = Veritrans.new(
+      server_key: "SB-Mid-server-uQmMImQMeo0Ky3Svl90QTUj2",
+      client_key: "SB-Mid-client-ArNfhrh7st9bQKmz",
+      api_host: "https://api.sandbox.midtrans.com",
+      logger: Logger.new(STDOUT),
+      file_logger: Logger.new(STDOUT)
+    )
+  end
+
+  def test_get_token_credit_card
+    card = {
+      card_number: 4811111111111114,
+      card_cvv: 123,
+      card_exp_month: 12,
+      card_exp_year: 2025
+    }
+    result = @mt_test.test_token(card)
+    assert_equal 200, result.status_code
+    assert_equal "Credit card token is created as Token ID.", result.status_message
+  end
+
+  def test_charge_credit_card
+    card = {
+      card_number: 5211111111111117,
+      card_cvv: 123,
+      card_exp_month: 12,
+      card_exp_year: 2025
+    }
+    get_token = @mt_test.test_token(card)
+    result = @mt_test.charge(
+      {
+        "payment_type": "credit_card",
+        "transaction_details": {
+          "gross_amount": 10000,
+          "order_id": "test-order-#{rand(1..10000)}"
```

Comment:
Becareful with `rand(1..10000` you may encounter order id duplicate after a while. It is better to use unix timestamp (better if the unix timestamp is accurate to exact milisecond), like this:
https://github.com/Midtrans/midtrans-nodejs-client/blob/b25bfd2e7364900893d515b24f3768c5fe1ea1c1/test/snap.test.js#L222

- if timestamp still produce duplicate order id, then add random digit suffix.
- Also make the order id prefix have clear indication it is coming from ruby lib test.